### PR TITLE
Fix explain analyze printing "never executed" uncorrectly

### PR DESCRIPTION
**Pull request by fanfuxiaoran**
explain analyze insert into test select i from
generate_series(0,100) as i;

Insert on test (never executed)
    ->  Result (actual time=0.021..0.046 rows=38 loops=1)
	    ->  Result (actual time=0.021..0.041 rows=38 loops=1)
		    ->  Function Scan on generate_series (actual
			    time=0.018..0.026 rows=101 loops=1)

As shown above, insert node is considered never executed. Actually, it has been executed. This is due to the collected nloops of that node is wrong. When the nloops in the instrument is 0, the plan node will be considered as 'never executed'.

In some plan nodes, the rows returned is 0, but the nloops is not, fix related code in gp_explain.c

fix #ISSUE_Number

---

### Change logs

_Describe your change clearly, including what problem is being solved or what feature is being added._

_If it has some breaking backward or forward compatibility, please clary._

### Why are the changes needed?

_Describe why the changes are necessary._

### Does this PR introduce any user-facing change?

_If yes, please clarify the previous behavior and the change this PR proposes._

### How was this patch tested?

_Please detail how the changes were tested, including manual tests and any relevant unit or integration tests._

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to request `cloudberrydb/dev` team for review and approval when your PR is ready🥳

### CODE REVIEW

File: src/backend/commands/explain_gp.c [hunk 1]
```diff
@@ -1191,24 +1196,33 @@ cdbexplain_depositStatsToNode(PlanState *planstate, CdbExplain_RecvStatCtx *ctx)
 	 * We don't print those "Rows Removed by Filter" rows in GPDB, because
 	 * they don't come from the "winner" QE.
 	 */
+	 /*
+	  * Alough the ntuples is 0, the nloops may be not, as the node
+	  * may has been executed without retruning any data
```
Comment (my-ship-it) [inline src/backend/commands/explain_gp.c:1201]: s/retruning/returning/ ?

**Comment by my-ship-it** [comment]
Big diffs between ans files. We need to fixing pax/vec cases after cherry-picking to Lightning. @gfphoenix78 @Light-City

File: src/backend/commands/explain_gp.c (hunk 1 above)
Comment (fanfuxiaoran) [reply src/backend/commands/explain_gp.c]: Fixed

**Comment by my-ship-it** [comment]
Please fix CI failures

**Comment by fanfuxiaoran** [comment]
> Please fix CI failures

Looks like the failures are related to ao vacuum, will fix them.

**Comment by fanfuxiaoran** [comment]
> Please fix CI failures

Have tested locally and rerun it on the ci, cannot reproduce the failed case.

**Pull request merged**